File: core/SwapNodesInPairs/SwapNodesInPairs.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def swapPairs(head: ListNode) -> ListNode:
    pre_head = ListNode(0)
    pre_head.next = head
    cur_node = pre_head
    while cur_node.next and cur_node.next.next:
        next_node = cur_node.next
        temp_node = cur_node.next.next.next
        cur_node.next = next_node.next
        cur_node.next.next = next_node
        next_node.next = temp_node
        cur_node = next_node

    return pre_head.next

File: core/SwapNodesInPairs/SwapNodesInPairs.java
package com.dbc;

public class SwapNodesInPairs {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode swapPairs(ListNode head) {
        ListNode preNode = new ListNode();
        preNode.next = head;
        ListNode curNode = preNode;
        while (curNode.next != null && curNode.next.next != null) {
            ListNode nextNode = curNode.next;
            ListNode tempNode = curNode.next.next.next;
            curNode.next = nextNode.next;
            curNode.next.next = nextNode;
            nextNode.next = tempNode;
            curNode = nextNode;
        }
        return preNode.next;
    }
}
